File: ios/Sources/BIShareProtocol/Constants/CryptoConstants.swift
import Foundation

/// Cryptographic constants for BIShare E2E encryption.
public enum BIShareCrypto {
    // MARK: - E2E File Transfer (Curve25519 ECDH)

    /// HKDF salt for local E2E file transfer
    public static let e2eSalt = "BIShare-E2E"
    /// HKDF info for local E2E file transfer
    public static let e2eInfo = "file-transfer"

    // MARK: - Remote Transfer (symmetric key from share code)

    /// HKDF salt for remote transfer via relay
    public static let remoteSalt = "BIShare-Remote"
    /// HKDF info for remote transfer via relay
    public static let remoteInfo = "remote-transfer-v2"

    // MARK: - AES-256-GCM Parameters

    /// AES key size in bytes (256 bits)
    public static let aesKeySize = 32
    /// GCM nonce size in bytes
    public static let gcmNonceSize = 12
    /// GCM authentication tag size in bits
    public static let gcmTagBits = 128

    // MARK: - Key Fingerprint

    /// Number of SHA-256 hash bytes used for visual fingerprint
    public static let fingerprintBytes = 8
}

File: android/src/main/java/com/bishare/protocol/constants/CryptoConstants.kt
package com.bishare.protocol.constants

/**
 * Cryptographic constants for BIShare E2E encryption.
 */
object BIShareCrypto {
    // E2E File Transfer (X25519 ECDH)
    const val E2E_SALT = "BIShare-E2E"
    const val E2E_INFO = "file-transfer"

    // AES-256-GCM Parameters
    /** AES key size in bytes (256 bits) */
    const val AES_KEY_SIZE = 32
    /** GCM nonce size in bytes */
    const val GCM_NONCE_SIZE = 12
    /** GCM authentication tag size in bits */
    const val GCM_TAG_BITS = 128

    // Key Fingerprint
    /** Number of SHA-256 hash bytes used for visual fingerprint */
    const val FINGERPRINT_BYTES = 8

    // Streaming Encryption (v2.2)

    /** GCM overhead per encrypted chunk: nonce(12) + tag(16) */
    const val GCM_OVERHEAD_PER_CHUNK = GCM_NONCE_SIZE + (GCM_TAG_BITS / 8)
}
